packages/opentelemetry-sdk-trace-base/src/Span.ts

```diff
@@ -272,8 +272,19 @@ export class Span implements APISpan, ReadableSpan {
     return this._ended === false;
   }
 
-  recordException(exception: Exception, time?: TimeInput): void {
-    const attributes: SpanAttributes = {};
+  recordException(
+    exception: Exception,
+    attributesOrStartTime?: SpanAttributes | TimeInput,
+    timeStamp?: TimeInput
+  ): void {
+    if (isTimeInput(attributesOrStartTime)) {
```

Comment:
Why not adding attributes as last parameter to avoid casting here?, attributes should be optional as well according to spec

Reply:
The attributes are optional already. I've kept the method signature and the code completely symmetrical to the [`addEvent`](https://github.com/open-telemetry/opentelemetry-js/blob/b400c2e5d9729c3528482781a93393602dc6dc9f/packages/opentelemetry-sdk-trace-base/src/Span.ts#L187-L194). Since these are related APIs, I think it'd benefit to keep them as much as possible.